include/deal.II/grid/grid_tools.h

```diff
@@ -1285,6 +1285,269 @@ namespace GridTools
       const bool                                             perform_handshake,
       const bool enforce_unique_mapping = false);
 
+
+    /**
+     * A function that fills DistributedComputeIntersectionLocationsInternal.
+     * @p intersection_requests are vertices of cells which describe the
+     * entities we want to search intersections for.
+     * The template parameter @p request_dim provides the dimension of the
+     * requesting intersection, e.g., if the intersection request corresponds
+     * to faces request_dim = dim-1 and for cells request_dim = dim.
+     */
+    template <int request_dim,
+              int dim,
+              int spacedim,
+              typename IntersectionRequest>
+    DistributedComputeIntersectionLocationsInternal<dim,
+                                                    spacedim,
+                                                    IntersectionRequest>
+    distributed_compute_intersection_locations(
```

Comment:
Could you move the implementation to `.cc`?

Reply:
If we instantiate certain ```IntersectionRequestTypes``` only this should work. If not, we can not shift it. Since we also have ```structdim``` we might be able to generate the ``IntersectionRequestType``` internally. I think we should aim for this solution.

Reply:
If we instantiate certain IntersectionRequestTypes only this should work. If not, we can not shift it. Since we also have structdim we might be able to generate the ``IntersectionRequestType``` internally. I think we should aim for this solution.

Reply:
Already done in Same discussion in https://github.com/dealii/dealii/pull/15165